openedx-unsupported/edx-analytics-pipeline
# implement adyen payment reporting

SOL-1880
@hasnain-naveed @ibrahimahmed443 @douglashall 
- Implement `Adyen` daily payment reporting
- Add tests config section for `Adyen`
- Add unit tests for `Adyen` reporting
- Update payment acceptance tests for `Adyen` reports
### How to run this locally
- Start your analytics devstack http://edx.readthedocs.io/projects/edx-installing-configuring-and-running/en/latest/installation/analytics/start_analytics.html
- Add settings for `adyen` to your `edx-analytics-pipeline/config/devstack.cfg`

```
[adyen]
company_account = TestAccount
merchant_accounts_list = EdXORG
username = report
password = ************
url = https://ca-test.adyen.com/reports/download
interval_start = 2016-08-19
```
- Run the remote command against your `analytics devstack` from your `edx-analytics-pipeline` for a specific day for example to get report for `2016-08-17` run following command:

``` shell
remote-task --vagrant-path ~/Workspace/analytics-edx/analyticstack/ --remote-name devstack --override-config ${PWD}/config/devstack.cfg --wheel-url $WHEEL_URL --branch origin/zub/sol-1880-implement-adyen-transactions-reporting --wait DailyProcessFromAdyenTask --company-account EdXAccount --merchant-account EdXORG --output-root hdfs://localhost:9000/edx-analytics-pipeline/output/adyen/ --run-date=2016-08-17 --local-scheduler
```
- SSH your `analytics` devstack
- Switch to `hadoop` user

```
sudo su hadoop
```
- Verify that there is respective report available by running this command:

```
hadoop@precise64:/home/vagrant$ hadoop fs -ls /edx-analytics-pipeline/output/adyen/
Found 1 items
drwxr-xr-x   - hadoop supergroup          0 2016-08-31 10:32 /edx-analytics-pipeline/output/adyen/payments
```
